### ISSUE TITLE
"attributes" belonging, when creating VO

### ISSUE THREAD

**Issue by pawski** [sabre-io/xml#137]
Hey, I was checking functionality of creating value object instead of array from xml.
I came to the point where I have doubts if xml node attributes are attached to correct place.
In short, attributes are not assigned to VO which is created for particular node, but to it's parent.

Please check modified unit test.
https://github.com/pawski/xml/pull/1/files
The output is:
```
1) Sabre\Xml\Deserializer\ValueObjectTest::testDeserializeValueObject
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
     'name' => '{urn:person}person'
     'value' => Sabre\Xml\Deserializer\PersonVo Object (...)
     'attributes' => Array (
+        'sex' => 'male'
     )
 )
```
I would expect attributes inside PersonVo, not "outside".
```
<<<XML
<?xml version="1.0"?>
<person xmlns="urn:person" sex="male">
   <firstName>Harry</firstName>
   <lastName>Turtle</lastName>
</person>
XML
```

Do You have any reason why attributes are treated like this?

**Comment by evert** [comment]
The reason is a bit complicated, but ultimately it's because the 'value' of an element is disconnected from the parent that contains it.

Your XML document only contains a `person` element, but this makes more sense with XML bodies with deeper structures. One way to look at it, is that every XML document as an invisible 'root' element, and that element contains 1 child element, the person element.

I would now recommend users use the `Service` class instead. It strips the top-level name, attribute and value when you use the `parse` function, which is most of the time what you'll want to do anyway. Optionally you can use the `expect` function instead, which allows you to make sure an exception gets thrown if it wasn't (one of the) root elements you expected.

That takes care of getting rid of the `attributes` from the top-level array, but it doesn't put it in the actual PersonVO class.

The `ValueObject` parsers _only_ deal with elements, and there's a reason for that. A lot of the parsers allow bi-directional usage. You can use them for both parsing, and generating new XML.

I didn't think there was a sane, default way to automatically map both attributes and child elements, so the choice was to provide a good default (child elements) and force users to write their own deserializers if they need anything more complex/special.

So if you want a `sex` property on PersonVO, just write a deserializer for it. It's probably easier than you think. 

The [sabre/xml-atom](https://github.com/sabre-io/xml-atom) package can serve as an example of how to do that. The main purpose of that repo is pretty much a demonstration for sabre/xml and show some best practices.

tl;dr: Ignore the `attributes` from the top-level array. You only care about the value. To get the attribute in your value, write a simple custom deserializer.

Once you have your deserializer, feel free to share it here. Happy to see if it could be written simpler.

**Comment by pawski** [comment]
Thank You for extensive explanation! :) I will digest it and respond afterwords.
